x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_model_health.tsx

```diff
@@ -87,26 +112,51 @@ export const TrainedModelHealth: React.FC<TrainedModelHealthProps> = ({
     tooltipText: React.ReactNode;
   };
   switch (modelState) {
-    case TrainedModelState.Started:
-      modelHealth = {
-        healthColor: 'success',
-        healthText: modelStartedText,
-        tooltipText: modelStartedTooltip,
+    case TrainedModelState.NotDeployed:
+    case MlModelDeploymentState.NotDeployed:
```

Comment:
Accommodating for the more broader spectrum of status values of `MlModelDeploymentState` while maintaining backward compatibility with other components that still use `TrainedModelState`.